src/lib/components/Modal/README.mdx

```diff
@@ -74,13 +114,433 @@ See [API](#api) for all available options.
   tells better what happens rather than “OK”.
 
 - Modal **automatically focuses the first form field** by default which allows
-  users to confirm the modal by hitting enter key (requires an action of type
-  `submit`). When no field is found then the first button in footer is focused.
-  To turn this feature off, set the `autofocus` prop to `false`.
+  users to confirm the modal by hitting enter key. When no field is found then
+  the primary button (in footer) is focused. To turn this feature off, set the
+  `autofocus` prop to `false`.
 
 - **Avoid stacking** of modals. While it may technically work, the modal is just
   not designed for that.
 
+## Composition
+
+Modal is decomposed into following components:
+
+- [Modal](#api)
+  - [ModalHead](#modalhead-1)
+    - [ModalCloseButton](#modalclosebutton)
+  - [ModalBody](#modalbody-1)
+    - [ModalContent](#modalcontent)
+    (may be wrapped with [ScrollView](/components/scroll-view))
+  - [ModalFooter](#modalfooter-1)
+
+Using different combinations, you can compose different kinds of modals,
+e.g. dialog modal, blocking modal, scrollable modal etc.
+
+<Playground>
+  {() => {
+    const [modalOpen, setModalOpen] = React.useState(null);
+    const modalPrimaryButtonRef = React.useRef();
+    const modalCloseButtonRef = React.useRef();
+    return (
+      <>
+        <Button
+          label="Launch blocking modal without title"
+          onClick={() => {
+            setModalOpen(1);
+            setTimeout(() => setModalOpen(null), 2500);
```

Comment:
It would require setting up interval which would update state with countdown, so the example would be even more messy. I would expect that 2,5 sec is short enough to not to implement something like that. What @mbohal thing about it?

Reply:
I agree with @bedrich-schindler an example that is too complicated is a bad example.

Reply:
OK